lib/client/doc.js

```diff
@@ -889,7 +900,7 @@ Doc.prototype._hardRollback = function(err) {
   this.fetch(function() {
     var called = op && callEach(op.callbacks, err);
     for (var i = 0; i < pending.length; i++) {
-      callEach(pending[i].callbacks, err);
+      called = callEach(pending[i].callbacks, err) || called;
```

Comment:
@nateps @ericyhwang do you know if this change will have any weird side-effects? Originally it would not count a pending callback as having called something, so it would both call back, _and_ emit, which seems a bit odd?

It's forced some updates to a couple of edge case tests, which I'll highlight also. I'm assuming they're okay to change...? In them, we actually assert that the callback is called _and_ an error is emitted; is that the expected behaviour?